tests/pytest_framework/src/syslog_ng/syslog_ng.py

```diff
@@ -26,9 +26,10 @@
 
 
 class SyslogNg(object):
-    def __init__(self, logger_factory, instance_paths, syslog_ng_ctl):
+    def __init__(self, logger_factory, instance_paths, syslog_ng_ctl, valgrind_usage):
```

Comment:
Adding valgrind_usage to the SyslogNg constructor looks strange to me, which I would like to avoid.

As I see:
We have 3 objects now: SyslogNg, SyslogNgCtl, and SyslogNgCli.

SyslogNg needs SyslogNgCli, because the start, stop etc messages needs to be forwarded to syslog-ng cli. But SyslogNg does not need SyslogNgCtl.

SyslogNgCli needs SyslogNgCtl, because some commands need to use the control socket.

So my proposal is, we should inject `syslog_ng_cli` instead of `syslog_ng_ctl` to `SyslogNg`. So syslog_ng_cli would be created `__register_instance`. Creating syslog_ng_ctl, should be moved into the constructor of `SyslogNgCli` from `__register_instance`. The `new_syslog_ng_ctl` should be removed.

valgrind_usage would be parameter of the syslog_ng_cli constructor instead. But I still feel this boolean parameter too specific, instead I would inject the whole `testcase_parameters` variable, and read the valgrind state through that.

Reply:
Ok, agreed with the changing the parameter list of SyslogNg, SyslogNgCtl, and SyslogNgCli classes. 
Also agreed with removing syslog_ng_ctl object from the hashmap in register_instance() .
Also ok for passing testcase_parameters instead of a specific value.